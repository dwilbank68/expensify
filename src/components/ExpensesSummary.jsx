import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import totalExpenses from '../selectors/expenseTotal';
import getVisibleExpenses from '../selectors/expenseFunctions';

import numeral from 'numeral';

// import ExpensesSummary from './ExpensesSummary.jsx';
// const ExpensesSummary = (props) => {
export
const ExpensesSummary = ({expenseCount, expenseTotal}) => {

    const units = expenseCount === 1 ? 'expense' : 'expenses';
    const expenseTotalFormatted = numeral(expenseTotal/100).format('$0,0.00')

    return (
        <div className="expenses-summary">
            <pre>
                <h1>
                    Viewing {expenseCount} {units} totalling {expenseTotalFormatted}
                </h1>
            </pre>
        </div>
    );
};

const mapStateToProps = state => {
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expenseTotal: totalExpenses(visibleExpenses)
    }
};


// ExpensesSummary.defaultProps = {};
// ExpensesSummary.propTypes = {
//     name:        PropTypes.string.isRequired,
//     id:          PropTypes.oneOfType([ PropTypes.number, PropTypes.string ]).isRequired,
//     message:     PropTypes.shape({ title: PropTypes.string, text: PropTypes.string }).isRequired,
//     comments:    PropTypes.arrayOf(React.PropTypes.object),
//     date:        PropTypes.instanceOf(Date)
// };
//
// PropTypes -> array, bool, func, number, object, string, symbol

// export default ExpensesSummary;
export default connect(mapStateToProps)(ExpensesSummary);