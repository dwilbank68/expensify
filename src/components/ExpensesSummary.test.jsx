import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from './ExpensesSummary';

import expenses from '../tests/fixtures/expenses';

test('should render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(
        <ExpensesSummary    expenseCount={1}
                            expenseTotal={9500}/>
    )
    expect(wrapper).toMatchSnapshot();
})

test('should render ExpensesSummary with 2 expenses', () => {
    const wrapper = shallow(
        <ExpensesSummary    expenseCount={2}
                            expenseTotal={19500}/>
    )
    expect(wrapper).toMatchSnapshot();
})