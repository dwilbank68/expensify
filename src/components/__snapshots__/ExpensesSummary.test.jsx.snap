// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
<div
  className="expenses-summary"
>
  <pre>
    <h1>
      Viewing 
      1
       
      expense
       totalling 
      $95.00
    </h1>
  </pre>
</div>
`;

exports[`should render ExpensesSummary with 2 expenses 1`] = `
<div
  className="expenses-summary"
>
  <pre>
    <h1>
      Viewing 
      2
       
      expenses
       totalling 
      $195.00
    </h1>
  </pre>
</div>
`;
